core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -388,7 +388,7 @@ class Partition(val topic: String,
     val leaderHWIncremented = inWriteLock(leaderIsrUpdateLock) {
       leaderReplicaIfLocal match {
         case Some(leaderReplica) =>
-          val outOfSyncReplicas = getOutOfSyncReplicas(leaderReplica, replicaMaxLagTimeMs)
+          val outOfSyncReplicas = getOutOfSyncReplicas(leaderReplica, replicaMaxLagTimeMs).filter(r => !r.ignoreIsrCheck)
```

Comment:
@lindong28 : Dong, since you added this logic, do you think the above suggested approach works? If so, do you have any insight on why it may cause some existing tests to break?